hello-proxy-impl/src/main/scala/com/example/helloproxy/impl/HelloProxyServiceImpl.scala

```diff
@@ -10,15 +11,20 @@ import scala.concurrent.{ ExecutionContext, Future }
 /**
   * Implementation of the HelloStreamService.
   */
-class HelloProxyServiceImpl(helloService: HelloService, greeterClient: GreeterServiceClient)(implicit exCtx: ExecutionContext) extends HelloProxyService {
+class HelloProxyServiceImpl(
+                             helloService: HelloService,
+                             greeterClient: RestartingClient[GreeterServiceClient])(implicit exCtx: ExecutionContext) extends HelloProxyService {
 
   def proxyViaHttp(id: String) = ServiceCall { _ =>
     val eventualString: Future[String] = helloService.hello(id).invoke()
     eventualString
   }
 
   def proxyViaGrpc(id: String) = ServiceCall { _ =>
-    val eventualString: Future[String] = greeterClient.sayHello(HelloRequest(id)).map(_.message)
+    val eventualString: Future[String] =
+      greeterClient.withClient(
+        _.sayHello(HelloRequest(id)).map(_.message)
+      )
```

Comment:
Using a `RestartingClient` has the implication that userland code no longer looks like a plain RPC and has some extra noise. This is not what lagom users are accustomed to (in Lagom everything _just works_ :TM:)